### PR TITLE
Bump Azure.Core.Expressions.DataFactory version to latest beta

### PR DESCRIPTION
**Pull request by JoshLove-msft**
# Description
In order to leverage the new types of Azure.Core.Expressions.DataFactory in code gen, bumping to the latest beta. This also required updating the Azure.Core dependency to the latest in order to avoid downgrade errors. It also required the following additional fixes:

- Updating PropertyReferenceTypeAttribute to support passing InternalPropertiesToInclude
- Including these internal properties in BuildProperties for SystemObjectTypes

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: test/TestProjects/MgmtMockAndSample/src/Generated/Models/MhsmNetworkRuleSet.Serialization.cs
Comment (JoshLove-msft) [inline test/TestProjects/MgmtMockAndSample/src/Generated/Models/MhsmNetworkRuleSet.Serialization.cs:62]: This (and other similar instances) should not be changing. We will need to figure out a way to avoid these false positives that are caused by DataFactorySecretString having only a single Value property.

File: src/AutoRest.CSharp/Mgmt/Decorator/TypeReferenceTypeChooser.cs
Comment (JoshLove-msft) [inline src/AutoRest.CSharp/Mgmt/Decorator/TypeReferenceTypeChooser.cs:45]: This was necessary to avoid false positive replacements. If we need a more robust attribute-based solution we can investigate that as well, but this should at least unblock the DataFactory release.

File: test/AutoRest.TestServerLowLevel.Tests/httpInfrastructure.cs
Comment (archerzz) [inline test/AutoRest.TestServerLowLevel.Tests/httpInfrastructure.cs:172]: After merge of #3570, those test cases should work now.

**Review by m-nash — APPROVED** [review]
As long as the final regen preview testing goes well this is ready to merge

**Pull request merged**
